diff --git a/src/rp2_common/hardware_flash/flash.c b/src/rp2_common/hardware_flash/flash.c
index 620be4ca..5365634a 100644
--- a/src/rp2_common/hardware_flash/flash.c
+++ b/src/rp2_common/hardware_flash/flash.c
@@ -554,7 +554,7 @@ static void __no_inline_not_in_flash_func(flash_put_get)(const uint8_t *tx, uint
 }
 #endif
 
-static inline void flash_wait_ready(uint cs)
+static void __no_inline_not_in_flash_func(flash_wait_ready)(uint cs)
 {
 	uint8_t status_reg;
 #if PICO_RP2040
@@ -573,7 +573,7 @@ static inline void flash_wait_ready(uint cs)
 	} while (status_reg & 0x1 && !flash_was_aborted());
 }
 
-static inline void flash_enable_write(uint cs)
+static void __no_inline_not_in_flash_func(flash_enable_write)(uint cs)
 {
 #if PICO_RP2350
 	qmi_hw->direct_tx = FLASHCMD_WRITE_ENABLE | QMI_DIRECT_TX_NOPUSH_BITS;
@@ -586,7 +586,7 @@ static inline void flash_enable_write(uint cs)
 #endif
 }
 
-static inline void flash_put_cmd_addr(uint8_t cmd, uint32_t addr)
+static void __no_inline_not_in_flash_func(flash_put_cmd_addr)(uint8_t cmd, uint32_t addr)
 {
 #if PICO_RP2350
 	addr = __builtin_bswap32(addr & ((1u << 24) - 1));
